### PR TITLE
fix: Fixes LengthException when persisting versioned entity using class table inheritance

### PR DESCRIPTION
**Pull request by kryptychon**
This is a fix for #9378 and a replacement for #9379.

Now there is a fix and a test case as well. I'm sorry that I took that long.

### CODE REVIEW

**Comment by greg0ire** [comment]
About https://github.com/doctrine/orm/pull/9379#issuecomment-1015852345:

Your commit message should look like this:

```
title shorter than 50 chars

Here, you should put a lengthy explanation wrapped at 72 chars
detailing what is broken, what you did and why what you did
fixes anything.

Fixes #9378 
Closes #9379 
```

**Comment by kryptychon** [comment]
Thanks for the feedback. I fixed the commit message.

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9378Test.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        
+        $entity             = new GH9378Employee();
+        $entity->firstName  = $firstName;
+        $entity->lastName   = $lastName;
+        $entity->employeeId = $employeeId;
+
+        $this->_em->persist($entity);
+        
+        /*
+         * Wthout the fix in Doctrine\ORM\Persisters\Entity\JoinedSubclassPersister
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9378Test.php]:
```suggestion
         * Without the fix in Doctrine\ORM\Persisters\Entity\JoinedSubclassPersister
```

**Comment by greg0ire** [comment]
The build is broken.

**Review by greg0ire — COMMENTED** [review]
Needs to be rebased on the current patch branch

**Pull request closed without merging**